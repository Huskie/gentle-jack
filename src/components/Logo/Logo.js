import React from 'react';
import PropTypes from 'prop-types';
import styled from 'react-emotion';
import { withTheme } from 'emotion-theming';

const StyledWrapper = styled('div')`
  font-weight: 700;
  letter-spacing: 2px;
  text-transform: uppercase;
  a {
    color: ${props => props.theme.colours.delta};
    text-decoration: none;
  }
`;

const BaseLogo = ({ label, isLink }) => (
  <StyledWrapper>
    {isLink ? <a href="/" title={label}>{label}</a> : <span>{label}</span>}
  </StyledWrapper>
);

BaseLogo.defaultProps = {
  isLink: false,
};

BaseLogo.propTypes = {
  label: PropTypes.string.isRequired,
  isLink: PropTypes.bool,
};

const Logo = withTheme(BaseLogo);

export { Logo };
