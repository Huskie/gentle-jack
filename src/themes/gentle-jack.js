const theme = {
  borderRadius: '4px',
  colours: {
    alpha: '#2742b8',
    beta: '#222',
    gamma: '#06f',
    delta: '#fff',
    epsilon: '#1b6cff',
    zeta: '#ffd561',
    eta: '#4048ef',
  },
  gradients: {
    alpha: '#4048ef, #2742b8',
  },
  spacing: {
    componentMargin: '16px',
  },
  sizing: {
    maxWidth: '1080px',
  },
};

export { theme };
