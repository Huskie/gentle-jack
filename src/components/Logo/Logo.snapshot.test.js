import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'emotion-theming';

import { Logo } from './Logo';
import { theme } from '../../themes/gentle-jack';

describe('Logo snapshots:', () => {
  it('component renders with defaults', () => {
    const tree = renderer.create(
      <ThemeProvider theme={theme}>
        <Logo label="A test label" />
      </ThemeProvider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('component renders with all props set', () => {
    const tree = renderer.create(
      <ThemeProvider theme={theme}>
        <Logo isLink label="A test label" />
      </ThemeProvider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
