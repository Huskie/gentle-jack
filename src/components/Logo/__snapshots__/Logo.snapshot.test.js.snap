// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo snapshots: component renders with all props set 1`] = `
.emotion-0 {
  font-weight: 700;
  -webkit-letter-spacing: 2px;
  -moz-letter-spacing: 2px;
  -ms-letter-spacing: 2px;
  letter-spacing: 2px;
  text-transform: uppercase;
}

.emotion-0 a {
  color: #fff;
  -webkit-text-decoration: none;
  text-decoration: none;
}

<div
  className="emotion-0"
>
  <a
    href="/"
    title="A test label"
  >
    A test label
  </a>
</div>
`;

exports[`Logo snapshots: component renders with defaults 1`] = `
.emotion-0 {
  font-weight: 700;
  -webkit-letter-spacing: 2px;
  -moz-letter-spacing: 2px;
  -ms-letter-spacing: 2px;
  letter-spacing: 2px;
  text-transform: uppercase;
}

.emotion-0 a {
  color: #fff;
  -webkit-text-decoration: none;
  text-decoration: none;
}

<div
  className="emotion-0"
>
  <span>
    A test label
  </span>
</div>
`;
